src/Shared/FileUtilities.GetFolderPath.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Build.Shared
+{
+    internal static partial class FileUtilities
+    {
+        //  .NET Core doesn't currently have APIs to get the "special" folders (ie the ones defined in the Environment.SpecialFolder enum)
+        //  The code here is mostly copied out of the .NET reference source for this functionality
```

Comment:
Thanks for clarifying :+1: I know you're an MS employee so I just wanted to have something on record that clarifies this, licensing is hard :smile: